### ISSUE TITLE
Unable to create target link from link text

### ISSUE THREAD

**Issue by phil31753** [zsviczian/obsidian-excalidraw-plugin#431]
When I Alt-Click on the Excalidraw Link icon, intending to create a target link.

![Screenshot 2022-02-12 151131](https://user-images.githubusercontent.com/95326157/153701224-816733df-b29f-4f27-aa39-95f5073728fa.png)

I get a dialogue box displayed.

![Screenshot 2022-02-12 150504](https://user-images.githubusercontent.com/95326157/153701060-4d19c8b1-0783-45be-a380-fa83a631545f.png)

When I click on the "Create Markdon" or "Create Excalidraw" button, **nothing happens**.

If I click on the "Never Mind" button, the dialogue is closed and no link target is created.

**Comment by zsviczian** [comment]
I must have messed something up. I'll look at it tonight.

**Comment by zsviczian** [comment]
I just tested it it works for me. Do you see any errors in developer console (Ctl+Shift+I)?

Are you creating a valid wiki link? `[[link]]`

**Comment by zsviczian** [comment]
maybe the issue is with the button in the top (red circle on your screenshot). Try the native link icon above to text. Also, from the screenshot it seems to are not using the latest version... or have you set the native link button in settings fully transparent?

**Comment by phil31753** [comment]
Yes latest version with valid [[link]] and native link button **fully transparent**.

When I click on Create Markdown button I get error below in console:

```
Uncaught (in promise) TypeError: Cannot read property 'lastIndexOf' of undefined
    at Jt (app.js:1)
    at t.getNewFileParent (app.js:1)
    at t.<anonymous> (app.js:1)
    at app.js:1
    at Object.next (app.js:1)
    at app.js:1
    at new Promise (<anonymous>)
    at D (app.js:1)
    at t.createNewMarkdownFileFromLinktext (app.js:1)
    at HTMLButtonElement.o.onclick (eval at <anonymous> (app.js:1), <anonymous>:153:7261)
```
When I click on the Create Excalidraw button I get error below in the console:

```
Uncaught (in promise) TypeError: Cannot read property 'lastIndexOf' of undefined
    at Jt (app.js:1)
    at t.getNewFileParent (app.js:1)
    at t.<anonymous> (app.js:1)
    at app.js:1
    at Object.next (app.js:1)
    at app.js:1
    at new Promise (<anonymous>)
    at D (app.js:1)
    at t.createNewMarkdownFileFromLinktext (app.js:1)
    at HTMLButtonElement.u.onclick (eval at <anonymous> (app.js:1), <anonymous>:153:7458)
```

**Comment by zsviczian** [comment]
thanks. this is helpful. 

will need to do some trouble shooting...

do you have the latest version of Obsidian as well?

did you try with a different filename (both the link and the folder/name of the Excalidraw drawing?)

**Comment by zsviczian** [comment]
I found the error. I wonder why it worked for me at all :) I'll be releasing a fix a bit later today.

**Comment by phil31753** [comment]
Installed 1.6.9 but I still get the same error?

```
Uncaught (in promise) TypeError: t.lastIndexOf is not a function
    at Jt (app.js:1)
    at t.getNewFileParent (app.js:1)
    at t.<anonymous> (app.js:1)
    at app.js:1
    at Object.next (app.js:1)
    at app.js:1
    at new Promise (<anonymous>)
    at D (app.js:1)
    at t.createNewMarkdownFileFromLinktext (app.js:1)
    at HTMLButtonElement.t.createEl.onclick (eval at <anonymous> (app.js:1), <anonymous>:48:25691)
Jt @ app.js:1
t.getNewFileParent @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
(anonymous) @ app.js:1
D @ app.js:1
t.createNewMarkdownFileFromLinktext @ app.js:1
t.createEl.onclick @ VM121:48
async function (async)
t.createEl.onclick @ VM121:48
```

**Comment by zsviczian** [comment]
hm... I have tested this on two windows PCs, an iPad and on Android, and it works on all...
I am starting to run out of ideas. 
And everything else works? 
Does your drawing or the folder it is in, have any special characters in the filename?
Can you try turning off other plugins, restarting obsidian and then trying to create a new page from a link?

**Comment by zsviczian** [comment]
When you hover a link for which the file is missing, Obsidian offers to create the file by default. What happens if you click create file? Is the file created?

![image](https://user-images.githubusercontent.com/14358394/153852580-8759c135-8d20-47e6-9884-f97ec8a957cf.png)

**Comment by zsviczian** [comment]
I published another solution with 1.6.10. I am now using a different set of Obsidian functions to create the file. Hope this will now work for you 🤞

**Comment by phil31753** [comment]
Works OK now. But unfortunately, I updated the Excalidraw plugin before trying to disable plugins (which I did intended to do).  So it will be difficult to get to the bottom of why it wasn't working. (Maybe not so important as it works now?)

I also suspect plugins because I was getting the same problem on a Linux install of Obsidian running on a Chrome Book. Given that it worked on **all** your platforms, and **none** of mine, it points to something unique about my vault/install.

FYI I have the following plugins installed:

- Customizable page header
- Enhancing mindmap
- Kanban
- Linter
- Longform
- Wikilinks to MDLinks

Thanks for your effort - back to enjoying Excalidraw!
